### PR TITLE
Use Python:3-slim base image in Dockerfile

### PR DESCRIPTION
**Pull request by cjmayo**
Currently reduces image size from 896MB to 126MB.

### CODE REVIEW

**Comment by cjmayo** [comment]
Use https://github.com/linkchecker/linkchecker/archive/master.zip instead of git. Ubuntu git requires a number of dependencies to be installed.

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,10 @@
...
 
 RUN set -x \
-    && apt-get update -qq \
-    && apt-get install -y -qq --no-install-recommends git \
-    && pip install --no-cache-dir git+https://github.com/linkchecker/linkchecker.git@master \
-    && apt-get -y -qq purge git \
-    && apt-get autoremove -y -qq \
-    && apt-get clean -y -qq all \
+    && pip install --no-cache-dir https://github.com/linkchecker/linkchecker/archive/master.zip \
     && rm -rf /var/lib/apt/lists/* /usr/share/doc /usr/share/locale /usr/share/man
```
Comment (mgedmin) [inline Dockerfile:8]: The `rm -rf` is here to reduce the size of the docker image by removing things installed by apt-get in the same `RUN` command.

The `python:3-slim` base image has nothing in /var/lib/apt, /usr/share/locale, /usr/share/man, and removing the 1 megabyte of files in /usr/share/doc will only increase the size of our image due to the way Docker layers work.

**Review by mgedmin — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (cjmayo) [reply Dockerfile]: rm removed.

**Review by anarcat — COMMENTED** [review]
:+1:

**Pull request merged**